tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -595,6 +596,27 @@ const DeviceCommand EnqueueReadBufferCommand::assemble_device_command(uint32_t d
     command.set_consumer_cb_num_pages(consumer_cb_num_pages);
     command.set_num_pages(num_pages);
 
+    // Targeting fast dispatch on remote device means commands have to be tunneled through ethernet
+    bool route_through_ethernet = not device->is_mmio_capable();
+    if (route_through_ethernet) {
+        uint32_t router_cb_num_pages = get_consumer_data_buffer_size(true) / padded_page_size;
+        // TODO: UPDATE THIS BY MEASURING PERF OF DIFFERENT VALUES
+        if (router_cb_num_pages >= 4) {
+            router_cb_num_pages = (router_cb_num_pages / 4) * 4;
+            command.set_producer_router_transfer_num_pages(router_cb_num_pages / 4);
+            command.set_consumer_router_transfer_num_pages(router_cb_num_pages / 4);
```

Comment:
why divide/multiply/divide?  isn't the last divide just getting you back to where you started?